pinot-integration-tests/src/test/java/com/linkedin/pinot/integration/tests/QueryGenerator.java

```diff
@@ -799,7 +799,44 @@ public QueryFragment generatePredicate(String columnName) {
       return new StringQueryFragment(columnName + " BETWEEN " + leftValue + " AND " + rightValue);
     }
   }
+  /**
+   * Generator for single-value column <code>REGEX</code> predicate query fragment.
+   */
+  private class SingleValueRegexPredicateGenerator implements PredicateGenerator {
+    Random random = new Random();
 
+    @Override
+    public QueryFragment generatePredicate(String columnName) {
+      List<String> columnValues = _columnToValueList.get(columnName);
+      String value;
+      value = pickRandom(columnValues);
+      StringBuilder pqlRegexBuilder = new StringBuilder();
+      StringBuilder sqlRegexBuilder = new StringBuilder();
+      // do regex only for string type
+      if (value.startsWith("'") && value.endsWith("'")) {
+        // replace only one character for now with .* ignore the first and last character
+        int indexToReplaceWithRegex = 1 + random.nextInt(value.length() - 2);
+        for (int i = 1; i < value.length() - 1; i++) {
+          if (i == indexToReplaceWithRegex) {
+            pqlRegexBuilder.append(".*");
+            sqlRegexBuilder.append(".*");
+          } else {
+            pqlRegexBuilder.append(value.charAt(i));
+            sqlRegexBuilder.append(value.charAt(i));
+          }
+        }
+
+        String pql = String.format(" REGEXP_LIKE(%s, '%s')", columnName, pqlRegexBuilder.toString());
+        String sql = String.format(" REGEXP_LIKE(%s, '%s', 'i')", columnName, sqlRegexBuilder.toString());
+        System.out.println("Value:" + value + " pql:" + pql + " sql:" + sql);
```

Comment:
Do not do sout because it might polute the test log and break Travis-CI for the number of output lines limit.